pkg/apis/calico/v1alpha1/types_network.go

```diff
@@ -26,41 +26,59 @@ const (
 	VXLan Backend = "vxlan"
 )
 
-type IPv4PoolMode string
+type PoolMode string
 
 const (
-	Always      IPv4PoolMode = "Always"
-	Never       IPv4PoolMode = "Never"
-	CrossSubnet IPv4PoolMode = "CrossSubnet"
-	Off         IPv4PoolMode = "Off"
+	Always      PoolMode = "Always"
+	Never       PoolMode = "Never"
+	CrossSubnet PoolMode = "CrossSubnet"
+	Off         PoolMode = "Off"
 )
 
 type CIDR string
 
-type IPv4Pool string
+type Pool string
 
 const (
-	PoolIPIP  IPv4Pool = "ipip"
-	PoolVXLan IPv4Pool = "vxlan"
+	PoolIPIP  Pool = "ipip"
+	PoolVXLan Pool = "vxlan"
 )
 
 // IPv4 contains configuration for calico ipv4 specific settings
 type IPv4 struct {
 	// Pool configures the type of ip pool for the tunnel interface
 	// https://docs.projectcalico.org/v3.8/reference/node/configuration#environment-variables
 	// +optional
-	Pool *IPv4Pool `json:"pool,omitempty"`
+	Pool *Pool `json:"pool,omitempty"`
 	// Mode is the mode for the IPv4 Pool (e.g. Always, Never, CrossSubnet)
 	// ipip pools accept all pool mode values values
+	// TODO: VXLAN also supports CrossSubnet for VXLAN. Why is this not supported?
 	// vxlan pools accept only Always and Never (unchecked)
 	// +optional
-	Mode *IPv4PoolMode `json:"mode,omitempty"`
+	Mode *PoolMode `json:"mode,omitempty"`
 	// AutoDetectionMethod is the method to use to autodetect the IPv4 address for this host. This is only used when the IPv4 address is being autodetected.
 	// https://docs.projectcalico.org/v3.8/reference/node/configuration#ip-autodetection-methods
 	// +optional
 	AutoDetectionMethod *string `json:"autoDetectionMethod,omitempty"`
 }
 
+// IPv6 contains configuration for calico ipv6 specific settings
+type IPv6 struct {
+	// Pool configures the type of ip pool for the tunnel interface
+	// https://docs.tigera.io/calico/latest/reference/configure-calico-node#configuring-the-default-ip-pools
+	// +optional
+	Pool *Pool `json:"pool,omitempty"`
+	// Mode is the mode for the IPv6 Pool (e.g. Always, Never, CrossSubnet)
+	// TODO: VXLAN also supports CrossSubnet for VXLAN. Why is this not supported?
+	// vxlan pools accept only Always and Never (unchecked)
```

Comment:
Though, I don't know the answer to the TODO question, I would prefer not to have it here in the api-reference.
You could open an issue regarding the CrossSubnet support?

Reply:
Absolutely the TODO makes no sense here. I will create an issue about it